### PR TITLE
Simplify build process

### PR DESCRIPTION
**Pull request by npars**
- Remove dependencies on DevkitPro
- Use linker similarly to min-gba
- [x] Get tests working
- [x] Update Github Action
- [x] Fix debug build

### CODE REVIEW

**Comment by npars** [comment]
@Lokathor this should be good for review now

**Comment by Lokathor** [comment]
I'll try to check later today

File: README.md [hunk 1]
```diff
@@ -22,6 +22,26 @@ The following major GBA features are still missing from the crate:
...
+
+Install arm build tools
+* Ubuntu
+  ```shell
+  sudo apt-get install binutils-arm-none-eabi
+  ```
+* OSX
+  ```shell
+  brew install --cask gcc-arm-embedded
+  ```
```
Comment (Lokathor) [inline README.md:43]: This should also say something for windows. We can link to the download page on the ARM website: https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm/downloads

Comment (npars) [reply README.md:43]: Updated!

**Review by Lokathor — APPROVED** [review]

**Comment by Lokathor** [comment]
If you're done we can merge this.

**Comment by npars** [comment]
Go ahead, thanks!

**Pull request merged**